### PR TITLE
feat: adding support for checkboxgroup component

### PR DESCRIPTION
**Pull request by harshilgajera-crest**
Added support for testing checkbox-group component.

- Added support for testing  checkbox and textbox component using xpath rather then fixing it with css selector.
- Reused checkbox and textbox component inside checkbox-group component.
- Updated test cases to check the checkbox-group component

### CODE REVIEW

File: pytest_splunk_addon_ui_smartx/components/controls/checkboxgroup.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+            group_name (str): The name of the group to expand.
+            checkbox_name (str): The name of the checkbox to retrieve the value from.
+
+        Returns:
+            str: The value of the checkbox.
+        """
+        self.expand_group(group_name)
+        return self.get_textbox(checkbox_name).get_value()
+
+    def expand_group(self, group_name: str) -> None:
```
Comment (mkolasinski-splunk) [inline pytest_splunk_addon_ui_smartx/components/controls/checkboxgroup.py:128]: Does it make sense to add method collapse_group?

**Review by mkolasinski-splunk — COMMENTED** [review]
Consider covering all new functionalities with test scenarios.

File: pytest_splunk_addon_ui_smartx/components/controls/checkboxgroup.py (hunk 1 above)
Comment (harshilgajera-crest) [reply pytest_splunk_addon_ui_smartx/components/controls/checkboxgroup.py:128]: We can add but it as a functionality, but it won't block other browser interactions.
For eg: if a dropdown is open clicking other components may result in stale element or element not intractable error but not in this case.

Comment (mkolasinski-splunk) [reply pytest_splunk_addon_ui_smartx/components/controls/checkboxgroup.py:128]: Yes I do agree. But I consider expanding/collapsing as a part of the functionality which may/may not work. So how would we confirm, if pressing the collapse button, actually collapsed group section without such method here?
Or maybe it is covered in other scenario?

Comment (harshilgajera-crest) [reply pytest_splunk_addon_ui_smartx/components/controls/checkboxgroup.py:128]: Yes so expanding and those scenarios are covered in main functionality like select_checkbox_and_set_value.
For component level testing we can add that in UCC

File: tests/testdata/Splunk_TA_UCCExample/globalConfig.json [hunk 2]
```diff
@@ -1150,4 +1308,4 @@
         "displayName": "Splunk UCC test Add-on",
         "schemaVersion": "0.0.3"
     }
-}
+}
```
[the remark this answers is not in the document]
Comment (harshilgajera-crest) [reply tests/testdata/Splunk_TA_UCCExample/globalConfig.json]: done

File: pytest_splunk_addon_ui_smartx/components/controls/checkboxgroup.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        """
+        Initializes the CheckboxGroup.
+
+        Args:
+            browser (Browser): The browser instance to interact with.
+            container (Selector): The container element that holds the checkbox group.
+        """
+        super().__init__(browser, container)
+
+    def is_group_expanded(self, group_name: str) -> bool:
```
[the remark this answers is not in the document]
Comment (harshilgajera-crest) [reply pytest_splunk_addon_ui_smartx/components/controls/checkboxgroup.py]: Done

**Review by mkolasinski-splunk — APPROVED** [review]

**Pull request merged**